plugins/modules/zos_apf.py

```diff
@@ -0,0 +1,506 @@
+                    'status': ['preview'],
+                    'supported_by': 'comminuty'}
+
+DOCUMENTATION = r'''
+---
+module: zos_apf
+author:
+  - "Behnam (@balkajbaf)"
+short_description: Add/remove libraries to Authorized Program Facility (APF)
+description:
+  - Add/remove libraries to Authorized Program Facility (APF)
+  - Make APF statement persistent entries to data set or data set member
+  - Change APF list format to "DYNAMIC" or "STATIC"
+  - Get current APF list entries
+options:
+  dsname:
+    description:
+      - The name of z/OS data set (library) to be added/removed.
+    required: False
+    type: str
+    aliases: [ name, lib, library ]
+  state:
+    description:
+      - Ensure the library is added C(state=present) or removed C(state=absent)
+      - APF list format has to be "DYNAMIC"
+    required: False
+    type: str
+    choices:
+      - absent
+      - present
+    default: present
+  force_dynamic:
+    description:
+      - Ensure APF list format is "DYNAMIC" before add/remove libraries
+    required: False
+    type: bool
+    default: False
+  volume:
+    description:
+      - The volume identifier for the volume containing the library specified on
+        the C(dsname) parameter, value should be one of the following,
+        1. The volume serial number
+        2. Six asterisks (******), indicating that the system is to use the
+        volume serial number of the current system residence (SYSRES) volume.
+        3. *MCAT*, indicating that the system is to use the volume serial number
+        of the volume containing the master catalog.
+      - If C(volume) is not specified, C(dsname) has to be cataloged.
+    required: False
+    type: str
+  sms:
+    description:
+      - Indicates that the library specified on the C(dsname) parameter is
+        managed by the storage management subsystem (SMS), and therefore no
+        volume is associated with the library.
+      - If C(sms=True), C(volume) value will be ignored.
+    required: False
+    type: bool
+    default: False
+  operation:
+    description:
+      - Change AFP list format to "DYNAMIC" or "STATIC"
+      - Display AFP list current format
+      - Display APF list entries
+      - If C(operation!=None), add/remove operation will be ignored.
+      - If C(operation=list), C(dsname), C(volume) and C(sms) will be used
+        as filters
+    required: False
+    type: str
+    choices:
+      - set_dynamic
+      - set_static
+      - check_format
+      - list
+  persistent:
+    description:
+      - Add/remove persistent entries to/from a dataset
+      - C(dsname) will not be persisted/removed if C(persistent=None)
+    required: False
+    type: dict
+    suboptions:
+      persistds:
+        description:
+          - The dataset be used to Persist or Remove the APF entry
+        required: True
+        type: str
+      marker:
+        description:
+          - The marker line template.
```

Comment:
Do we have a max limit for the marker length to highlight here?

Reply:
added the limitation